File: ProjectEuler17/ProjectEuler17.java
import java.util.HashMap;

public class ProjectEuler17{
	public String numberInEnglish(int number){
		
		HashMap<Integer, String> ones = new HashMap<Integer, String>(){{
			put(1, "one");
			put(2, "two");
			put(3, "three");
			put(4, "four");
			put(5, "five");
			put(6, "six");
			put(7, "seven");
			put(8, "eight");
			put(9, "nine");
		}};
		
		HashMap<Integer, String> elevens = new HashMap<Integer, String>(){{
			put(10, "ten");
			put(11, "eleven");
			put(12, "twelve");
			put(13, "thirteen");
			put(14, "fourteen");
			put(15, "fifteen");
			put(16, "sixteen");
			put(17, "seventeen");
			put(18, "eighteen");
			put(19, "nineteen");
		}};
		
		HashMap<Integer, String> tens = new HashMap<Integer, String>(){{
			put(10, "ten");
			put(20, "twenty");
			put(30, "thirty");
			put(40, "forty");
			put(50, "fifty");
			put(60, "sixty");
			put(70, "seventy");
			put(80, "eighty");
			put(90, "ninety");
		}};
		
		if (number == 0){
			return "";
		}
		else if (number < 10){
			return ones.get(number);
		}
		else if (number < 20){
			return elevens.get(number);
		}
		else if (number < 100){
			return tens.get(number);
		}
		else{
			return "hundred";
		}
	}

	public static void main(String args[]){
		ProjectEuler17 call = new ProjectEuler17();
		int number = 1;
		int numberSum = 0;
		while (number < 1000){
			if (number < 20){
				numberSum += call.numberInEnglish(number).length();
			}
			else if (number < 100){
				numberSum += call.numberInEnglish(number - number % 10).length();
				numberSum += call.numberInEnglish(number % 10).length();
			}
			else{
				numberSum += call.numberInEnglish(number/100).length();
				numberSum += call.numberInEnglish(100).length();
				if (number % 100 >= 20){
					numberSum += call.numberInEnglish(number % 100 - number % 10).length();
					numberSum += call.numberInEnglish(number % 10).length();
				}
				else{
					numberSum += call.numberInEnglish(number % 100).length();
				}
				numberSum += 3;
				if (number % 100 == 0){
					numberSum -= 3;
				}
			}
			number += 1;
		}
		numberSum += "onethousand".length();
		System.out.println(numberSum);
	}
}

File: ProjectEuler17/ProjectEuler17.py
"""Created a dictionary to store the string representation of numbers that will be useful in the program"""

def numberInEnglish(number):
	ones = {1:'one', 2:'two', 3:'three', 4:'four', 5:'five', 6:'six', 7:'seven', 8:'eight', 9:'nine'}
	tens = {10:'ten', 20:'twenty', 30:'thirty', 40:'forty', 50:'fifty', 60:'sixty', 70:'seventy', 80:'eighty', 90:'ninety'}
	elevens = {10:'ten', 11:'eleven', 12:'twelve', 13:'thirteen', 14:'fourteen', 15:'fifteen', 16:'sixteen', 17:'seventeen', 18:'eighteen', 19:'nineteen'}
	hundred = {100:'hundred'}
	if number == 0:							#if the given number is 0, just return an empty string because we don't use zero when naming numbers
		return ''
	elif number < 10:						#for numbers less than 10, we can just return the string representation of numbers
		return ones[number]					
	elif number < 20:						#for numbers less than 20, since these are the special cases, we return from the dictionary accordingly
		return elevens[number]				 
	elif number < 100:						#for numbers less than hundred, we return the tens place first, then the ones place
		return tens[number]
	else:
		return hundred[number]				#for hundred, just return hundred and go back to tens and ones place accordingly

def printNumber():
	number = 1								#start the loop with number 1 as given in the question
	numberSum = 0							#to store the value of the sum
	while number < 1001:
		if number < 10:		#for numbers less than 10, we can just use the string representation of number and add the string length to number sum
			numberSum += len(numberInEnglish(number))
			print(numberInEnglish(number))
		elif number < 100:
			if number < 20:	#for numbers less than 20, we have special cases thus we handle it likewise
				numberSum += len(numberInEnglish(number))
				print(numberInEnglish(number))
			else:			#for numbers greater than 20, we first take the tens place and then the ones place
				numberSum += len(numberInEnglish(number - number % 10))	#for the tens place
				numberSum += len(numberInEnglish(number%10))			#for the ones place
				print(numberInEnglish(number - number % 10))
				print(numberInEnglish(number%10))
		elif number < 1000:	#for numbers less than 1000
			numberSum += len(numberInEnglish(number//100))	#for the hundreds place
			numberSum += len(numberInEnglish(100))			#to add that 'hundred'
			if (number % 100 >= 20):	#for numbers in tens place greater than 20, we break it down to tens and ones place and add accordingly
				numberSum += len(numberInEnglish(number%100 - number%10))	#tens place
				numberSum += len(numberInEnglish(number%10))				#ones place
				print(numberInEnglish(number%100 - number%10))
				print(numberInEnglish(number%10))
			else:
				numberSum += len(numberInEnglish(number%100))	#for numbers less than 20, we use the special cases and add them up
				print(numberInEnglish(number%100))
			print(numberInEnglish(number//100))
			print(numberInEnglish(100))
			numberSum += 3										#to add value for 'and' that occurs in every number after 100
			if (number % 100 == 0):
				numberSum -= 3			#to subtract the extra 3 which comes from 'and' in cases of perfect hundreds such as 100, 200 etc
		else:
			numberSum += len("onethousand")	#to add the one thousand value to number sum
		number += 1

	return numberSum

print(printNumber())